core_v14/src/decoder/mod.rs

```diff
@@ -307,10 +313,45 @@ pub struct ExtrinsicSignature {
 	pub extensions: Vec<(String, Value)>,
 }
 
-fn decode_v4_signature<'a>(data: &mut &'a [u8], metadata: &Metadata) -> Result<ExtrinsicSignature, DecodeError> {
-	let address = <MultiAddress<AccountId32, u32>>::decode(data)?;
-	let signature = MultiSignature::decode(data)?;
-	let extensions = metadata
+fn decode_call_data<'data, 'meta>(
```

Comment:
I've been hesitant to make too many things `pub` at the moment becaue this V14 crate is still fairly heavily WIP, so I've been learning towards not committing too much to the public API, and making sure that we don't expose too many things going forwards and make it a little harder to keep hacking on without breaking APIs.

w.r.t the parsing, I'd sortof have assumed that the top level parser + scale_info would be enough (given some type and some bytes, you'll get a decoded `Value` back), but I'd be interested to hear about any cases where the sub parsers were useful on their own!

Reply:
In my experience, instead of being afraid to break APIs, you can just put anything that isn't really stable inside of a `unstable` module. That way people (like me) who are ok depending on unstable features can do so without patching your library. Anyone who cares about that can avoid the module. This, to me, is the best of both worlds.

Reply:
The problem with the top-level parsers is that you can't compose them yourself. If I have call-data inside of a larger blob, it's harder to use your parser (although I can pull it out of the LateEof error in this case). But, for example, I had to manually write `decode_call_data` here because I couldn't compose that with the other parsers. If I had access to that directly I may not have patched this library (although you could argue it was a good thing that the patch got upstreamed).